src/System.Management.Automation/FormatAndOutput/common/TableWriter.cs

```diff
@@ -207,16 +231,20 @@ internal void GenerateRow(string[] values, LineOutput lo, bool multiLine, ReadOn
 
             if (multiLine)
             {
-                string[] lines = GenerateTableRow(values, currentAlignment, lo.DisplayCells);
+                lines = GenerateTableRow(values, currentAlignment, lo.DisplayCells);
 
                 for (int k = 0; k < lines.Length; k++)
                 {
                     lo.WriteLine(lines[k]);
                 }
+
+                return lines;
             }
             else
             {
-                lo.WriteLine(GenerateRow(values, currentAlignment, dc));
+                string line = GenerateRow(values, currentAlignment, dc);
+                lo.WriteLine(line);
+                return new string[1]{line};
```

Comment:
It would be good if we can avoid creating the transient string arrays. How about add a `List<string>` parameter to `GenerateRow`, and directly add those strings to the passed in string list? Then we can avoid the unnecessary string arrays as well as the iteration of the returned string arrays in `GenerateHeader`.

Reply:
Most of the existing code uses string arrays, but I can modify my new code to now do that.

Reply:
@SteveL-MSFT I submitted a PR to `SteveL-MSFT:format-table-header` to propose a different way to organize this code: https://github.com/SteveL-MSFT/PowerShell/pull/5
The method `GenerateRow` is called from `ProcessPayload`, which will 1) run frequently; 2) not care about capturing the generated rows. So we should avoid creating a list every time the method is called. The proposed way can minimize list allocation and avoid a potential perf regression.

Reply:
Looks good.  Merged.